### PR TITLE
Adds audio querying to MultimodalQ&A gateway

### PR DESCRIPTION
**Pull request by mhbuehler**
## Description

Adds ASR endpoint, speech audio processing, prompt construction, and return of decoded audio in response metadata. This goes with GenAIExamples PR: https://github.com/opea-project/GenAIExamples/pull/1225.

## Issues

Part of the [MultimodalQnA Audio & Image Enhancements RFC](https://github.com/opea-project/docs/blob/main/community/rfcs/24-10-02-GenAIExamples-001-Image_and_Audio_Support_in_MultimodalQnA.md)

## Type of change

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

N/A

## Tests

Automated tests were added to GenAIExamples.

### CODE REVIEW

File: comps/cores/mega/gateway.py [hunk 1]
```diff
@@ -837,6 +838,9 @@ def parser_input(data, TypeClass, key):
 
 
 class MultimodalQnAGateway(Gateway):
+    asr_port = int(os.getenv("ASR_SERVICE_PORT", 3001))
+    asr_endpoint = os.getenv("ASR_SERVICE_ENDPOINT", "http://0.0.0.0:{}/v1/audio/transcriptions".format(asr_port))
```
Comment (ashahba) [inline comps/cores/mega/gateway.py:842]: Is `ASR_SERVICE_ENDPOINT` documented anywhere? I know the `ASR_SERVICE_PORT` is not new.
Or maybe you have it documented in the corresponding GenAIExamples PR.

Comment (mhbuehler) [reply comps/cores/mega/gateway.py:842]: It is in the corresponding GenAIExamples PR, e.g. here: https://github.com/opea-project/GenAIExamples/pull/1225/files#diff-1d1cfced259dada6f277c25da44cbaea85aa3629b3ce3fb54780c4525a9a3896R83, but if there's another place in this repo where it should be added please let me know.

**Review by ashahba — APPROVED** [review]
LGTM!

File: comps/cores/mega/gateway.py [hunk 2]
```diff
@@ -886,55 +903,84 @@ def _handle_message(self, messages):
...
+            b64_types["image"] = images
+
+        # If the query has multiple media types, return all types
+        if prompt and b64_types:
+            return prompt, b64_types
         else:
             return prompt
 
+    def convert_audio_to_text(self, audio):
+        # translate audio to text by passing in dictionary to ASR
```
Comment (mkbhanda) [inline comps/cores/mega/gateway.py:968]: comment quirky!  dictionary is a data type here but can get mixed with the English word dictionary (word meanings)

File: comps/cores/mega/gateway.py [hunk 3]
```diff
@@ -886,55 +903,84 @@ def _handle_message(self, messages):
...
+        if isinstance(audio, dict):
+            input_dict = {"byte_str": audio["audio"][0]}
+        else:
+            input_dict = {"byte_str": audio[0]}
+
+        response = requests.post(self.asr_endpoint, data=json.dumps(input_dict), proxies={"http": None})
```
Comment (mkbhanda) [inline comps/cores/mega/gateway.py:974]: should proxies be read from some environment variable for a more general solution?

**Review by mkbhanda — APPROVED** [review]
Looks good to me!

File: comps/cores/mega/gateway.py (hunk 3 above)
Comment (eero-t) [reply comps/cores/mega/gateway.py:974]: Why this is setting proxies in the first place, shouldn't those be set well before this point?

File: tests/cores/mega/test_multimodalqna_gateway.py [hunk 4]
```diff
@@ -2,15 +2,20 @@
...
 
 import json
+import os
 import unittest
 from typing import Union
 
 import requests
 from fastapi import Request
 
+os.environ["ASR_SERVICE_PORT"] = "8086"
```
Comment (eero-t) [inline tests/cores/mega/test_multimodalqna_gateway.py:12]: Why this overrides environment, instead of taking the value from environment?

**Comment by ashahba** [comment]
Labeling this as `WIP` since we may not need it after all.

**Comment by ashahba** [comment]
Looking at new changes in `GenAIExamples`, it seems like we don't this PR after all since this PR https://github.com/opea-project/GenAIExamples/pull/1225 is self contained.

**Pull request closed without merging**